onnxruntime/contrib_ops/cpu/matmul_integer16.cc

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "contrib_ops/cpu/matmul_integer16.h"
+#include "core/providers/cpu/math/matmul_helper.h"
+
+namespace onnxruntime {
+namespace contrib {
+
+ONNX_OPERATOR_KERNEL_EX(
+    MatMulInteger16,
```

Comment:
What is the use case for adding this op? Do we have a model which needs this? Can we add a cast operator and make it work with MatMul?

Reply:
There are 16-bit integer madd instructions like _mm256_madd_epi16, and the speed up is significant comparing to floating point.

---
In reply to: [303696295](https://github.com/microsoft/onnxruntime/pull/1391#discussion_r303696295) [](ancestors = 303696295)